### ISSUE TITLE
🐛 [a11y] - Breadcrumbs accessibility issues

### ISSUE THREAD

**Issue by jjwinskill** [primer/brand#819]
### Describe the bug

When a breadcrumb item takes the prop "selected", it is put in a non-interactive state signifying that the breadcrumb item is the same as the active page.

When in this state, a breadcrumb item has no hover state and cannot be clicked. However, it can still be focused and interacted with via keyboard navigation.

### Reproduction steps

```
1. Go to the Storybook implementation of Breadcrumbs at https://primer.style/brand/components/Breadcrumbs
2. Try interacting with the mouse on any bread crumb item in a "selected" state
3. Observe that nothing happens
4. Use the tab key to navigate through the each of the interactive elements, until the Breadcrumb items are reached
5. Using tab, see that the "selected" breadcrumb is focusable
6. When this breadcrumb item is focused, press the enter key to see that it is interactive
```

### Expected behavior

```
1. Go to the Storybook implementation of Breadcrumbs at https://primer.style/brand/components/Breadcrumbs
2. Try interacting with the mouse on any bread crumb item in a "selected" state
3. Observe that nothing happens
4. Use the tab key to navigate through the each of the interactive elements, until the Breadcrumb items are reached
5. Using tab, see that the "selected" breadcrumb is skipped and is not focusable
```

### Screenshots

**Mouse interaction works as expected**

https://github.com/user-attachments/assets/2fdb342e-cc98-4924-9a65-54edc5b2370e

**Keyboard interaction a11y bug**

https://github.com/user-attachments/assets/b3df0dec-e034-47b8-9b67-e1452957111b

### Browsers

Chrome

### OS

Mac
